packages/flutter_tools/lib/src/build_runner/web_fs.dart

```diff
@@ -501,7 +503,14 @@ class DebugAssetServer extends AssetServer {
       final String assetPath = request.url.path.replaceFirst('assets/', '');
       final File file = fs.file(fs.path.join(getAssetBuildDirectory(), assetPath));
       if (file.existsSync()) {
-        return Response.ok(file.readAsBytesSync());
+        final Uint8List bytes = file.readAsBytesSync();
+        // Fallback to back to "application/octet-stream" on null which
+        // makes no claims as to the structure of the data.
+        final String mimeType = mime.lookupMimeType(file.path, headerBytes: bytes)
```

Comment:
The argument seems to be expecting only the header bytes but we are sending the whole file. Is that not an issue?

Reply:
We're already reading the bytes so I was going to reuse the same buffer